Lib/email/_header_value_parser.py

```diff
@@ -2802,9 +2804,12 @@ def _refold_parse_tree(parse_tree, *, policy):
             wrap_as_ew_blocked -= 1
             continue
         tstr = str(part)
-        if part.token_type == 'ptext' and set(tstr) & SPECIALS:
+        if part.token_type == 'ptext' and set(tstr) & SPECIALSNL:
             # Encode if tstr contains special characters.
             want_encoding = True
+        elif set(tstr) & NLSET:
+            # Encode if text contains newlines
+            want_encoding = True
```

Comment:
This can be optimized:
```suggestion
        if part.token_type == 'ptext'
            # Encode if tstr contains special characters.
            want_encoding = not SPECIALSNL.isdisjoint(tstr)
        else:
            # Encode if text contains newlines
            want_encoding = not NLSET.isdisjoint(tstr)
```

Reply:
Hm, that changes the behaviour -- it can reset `want_encoding` to False. We don't want that.